dlpy/tests/test_image_captioning.py

```diff
@@ -0,0 +1,409 @@
+                               pre_trained_weights=True,
+                               pre_trained_weights_file=self.data_dir+'VGG_ILSVRC_16_layers.caffemodel.h5')
+        detection_model = Model(self.s)
+        detection_model.load(self.data_dir + 'YOLOV2_MULTISIZE.sashdat')
+        detection_model.load_weights(self.data_dir + 'YoloV2_Multisize_weights.sashdat')
+        word_embeddings = self.data_dir + 'word_embeddings.txt'
+        self.assertTrue(create_captioning_table(self.s,
+                                                  image_table,
+                                                  features_model,
+                                                  captions_file,
+                                                  obj_detect_model=detection_model,
+                                                  word_embeddings_file=word_embeddings) is not None)
+
+    # wrong image width object detection
+    def test_captioning_table_4(self):
+        if self.data_dir is None:
+            unittest.TestCase.skipTest(self, "DLPY_DATA_DIR is not set in the environment variables")
+
+        img_path = self.data_dir + 'imageCaptioning_images'
+        image_table = ImageTable.load_files(self.s,path=img_path)
+        image_table.resize(width=224)
+        captions_file = self.data_dir + 'image_captions.txt'
+        features_model = VGG16(self.s,
+                               width=224,
+                               height=224,
+                               pre_trained_weights=True,
+                               pre_trained_weights_file=self.data_dir+'VGG_ILSVRC_16_layers.caffemodel.h5')
+        detection_model = Model(self.s)
+        detection_model.load(self.data_dir + 'YOLOV2_MULTISIZE.sashdat')
+        detection_model.load_weights(self.data_dir + 'YoloV2_Multisize_weights.sashdat')
+        word_embeddings = self.data_dir + 'word_embeddings.txt'
+        self.assertRaises(SWATError,lambda:
+                                        create_captioning_table(self.s,
+                                                                image_table,
+                                                                features_model,
+                                                                captions_file,
+                                                                obj_detect_model=detection_model,
+                                                                word_embeddings_file=word_embeddings,
+                                                                objects_width=4000))
+
+    # works
+    def test_ImageCaptioning_1(self):
+        if self.data_dir is None:
+            unittest.TestCase.skipTest(self, "DLPY_DATA_DIR is not set in the environment variables")
+
+        model = ImageCaptioning(self.s,rnn_type='GRU')
+
+        model = ImageCaptioning(self.s,rnn_type='RNN')
+
+        model = ImageCaptioning(self.s)
+
+        model.print_summary()
+
+        self.assertTrue(model is not None)

```

Comment:
maybe you can check model1 = .... assert(model1 is not None), model2 = ...... assert(model2 is not None).